notes-with-indexeddb/notes.js

```diff
@@ -0,0 +1,169 @@
+}
+
+function createIndex(event) {
+    var db = event.target.result,
+        store = db.createObjectStore('notes', {
+            autoIncrement: true
+        });
+
+    db.onerror = showErrorMessage;
+
+    // create indices to search db by
+    store.createIndex('title', 'title', {
+        unique: false
+    });
+    store.createIndex('description', 'description', {
+        unique: false
+    });
+}
+
+/*
+populateDb with data
+ */
+function populateDb(db, data) {
+    data.forEach(function(row) {
+        addItem(db, row);
+    });
+}
+
+/*
+removeItem from db
+ */
+function removeItem(db, key) {
+    var request = db.transaction(['notes'], 'readwrite')
+        .objectStore('notes')
+        .delete(key);
+
+    request.onsuccess = function(event) {
+        showNotes(db);
+    };
+}
+
+function showNotes(db) {
+    var list = document.createDocumentFragment(),
+        notes = document.getElementById('notes');
+
+    /*
+    listNotes with cursor
+     */
+    function listNotes(event) {
+        var cursor = event.target.result;
+
+        if (cursor) {
+            var title = document.createElement('h3'),
+                description = document.createElement('p'),
+                item = document.createElement('div'),
+                button = document.createElement('button');
+
+            title.innerHTML = cursor.value.title;
+            description.innerHTML = cursor.value.description;
+
+            // delete action
+            button.setAttribute('id', cursor.primaryKey);
+            button.onclick = function(event) {
+                var key = +event.target.getAttribute('id');
+                removeItem(db, key);
+            };
+            button.innerHTML = 'Remove';
+
+            item.appendChild(title);
+            item.appendChild(description);
+            item.appendChild(button);
+
+            list.appendChild(item);
+
+            cursor.continue();
+        } else {
+            notes.innerHTML = '';
+            notes.appendChild(list);
+        }
+    }
+
+    var store = db.transaction('notes').objectStore('notes');
+
+    store.openCursor().onsuccess = listNotes;
+}
+
+function notesApp(event) {
+    const seedData = [{
+        title: 'To-do',
+        description: 'Buy cheese'
+    }, {
+        title: 'Books I like',
+        description: 'Peak: the science of expertise'
+    }];
+
+    var db = event.target.result;
+    db.onerror = showErrorMessage;
+
+    populateDb(db, seedData);
+    showNotes(db);
+
+    // add new item
+    var form = document.getElementById('add-new-item');
+    form.addEventListener('submit', function(event) {
+        var title = document.getElementById('new-title'),
+            description = document.getElementById('new-description');
+
+        addItem(db, {
+            title: title.value,
+            description: description.value
+        }, function() {
+            showNotes(db);
+            form.reset();
+        });
+
+        event.preventDefault();
```

Comment:
Please move this to the start of the event handler so that if there's an unexpected exception, it still gets called.